### PR TITLE
Dual ToR IO test enhancements

### PR DESCRIPTION
**Pull request by vaibhavhd**
### Description of PR

Summary: Dualtor I/O tests organized into per-link packet streams, and measure per-server traffic disruptions.
Fixes # (issue) https://github.com/Azure/sonic-mgmt/issues/3106

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Approach
Change the data plane I/O tests to send traffic from/to each server (28 servers as of now).
These changes are made in order to accurately measure packet loss and disruptions for individual links.
Changes:
1. Send downstream traffic to each server in round-robin fashion.
2. Send upstream traffic from each server in round-robin fashion.
3. Detect packet loss and collect disruptions data per-server (for every link).
4. Create a report per server.

#### What is the motivation for this PR?
Enhancements of dualtor data plane utils.

#### How did you verify/test it?
Tested on physical testbeds. Scenarios tested: Upstream/Downstream traffic with disruptive/non-disruptive actions.
Examples of non-disruptive test:
```
{
    "longest_disruption": 0,
    "longest_duplication": 0,
    "received_packet_diff": 0,
    "received_packets": 1500,
    "total_disruptions": 0,
    "total_duplications": 0
}
```

Traffic with disruptive action (CLI switchover):
```
{
    "longest_disruption": 0,
    "longest_duplication": 0.7100210189819336,
    "received_packet_diff": 9,
    "received_packets": 1509,
    "total_disruptions": 0,
    "total_duplications": 1
}
```
or
```
{
    "longest_disruption": 6.295104026794434,
    "longest_duplication": 0,
    "received_packet_diff": -69,
    "received_packets": 1431,
    "total_disruptions": 1,
    "total_duplications": 0
}
```

### CODE REVIEW

File: tests/common/dualtor/dual_tor_io.py [hunk 1]
```diff
@@ -51,22 +52,24 @@ def __init__(self, activehost, standbyhost, ptfhost, ptfadapter, tbinfo,
 
         self.time_to_listen = 180.0
         self.sniff_time_incr = 60
-        self.send_interval = 0.0035 # Inter-packet interval
+        # Inter-packet send-interval (minimum interval 3.5ms)
+        self.send_interval = send_interval if send_interval and send_interval > 0.0035 else 0.0035
```
Comment (theasianpianist) [inline tests/common/dualtor/dual_tor_io.py:56]: Can we add a warning log here in case the user passes a values < .0035?

Comment (vaibhavhd) [reply tests/common/dualtor/dual_tor_io.py]: Sure. Added now.

File: tests/common/dualtor/data_plane_utils.py [hunk 2]
```diff
@@ -69,16 +69,59 @@ def generate_test_report(tor_IO):
             "duplicated_packets_count": tor_IO.get_duplicated_packets_count(),
             "disruptions": {
                 "total_disruptions": tor_IO.get_total_disruptions(),
-                "total_disrupted_packets": tor_IO.get_total_disrupted_packets(),
-                "total_disruption_time": tor_IO.get_total_disrupt_time(),
                 "longest_disruption": tor_IO.get_longest_disruption(),
                 "total_lost_packets": tor_IO.get_total_lost_packets()
-            }
+            },
+            "per_server_report": tor_IO.get_per_server_report()
```
Comment (theasianpianist) [inline tests/common/dualtor/data_plane_utils.py:75]: Can we sort the results by server IP?

Comment (vaibhavhd) [reply tests/common/dualtor/data_plane_utils.py]: Change added in the latest

**Review by theasianpianist — APPROVED** [review]

**Pull request merged**